api4/user.go

```diff
@@ -44,6 +44,7 @@ func (api *API) InitUser() {
 	api.BaseRoutes.User.Handle("/password", api.ApiSessionRequired(updatePassword)).Methods("PUT")
 	api.BaseRoutes.User.Handle("/promote", api.ApiSessionRequired(promoteGuestToUser)).Methods("POST")
 	api.BaseRoutes.User.Handle("/demote", api.ApiSessionRequired(demoteUserToGuest)).Methods("POST")
+	api.BaseRoutes.User.Handle("/verify", api.ApiSessionRequired(verifyUser)).Methods("PUT")
```

Comment:
Why are creating a new endpoint having the `/email/verify`. I know the last one is used to verify by token but I want to know why are we verifying directly the email without token.

Reply:
Good question, I reckon that a system admin may want to create users and verify them. W/o the e-mail path. And even if it is not required anymore, we need it to maintain backwards compatibility.

Reply:
The permissions for the token verification are quite different, so I'm happy having these two endpoints separate as the use cases may differ a lot 2/5

Reply:
Yeah. My concern is about the naming of the endpoint. We have a `/email/verify` endpoint which should be this new one and we should have a `/email/verifytoken` that should be the one we already have.

Also `/user/UUID/verify` is a bit misleading. I'd go for something like `/user/UUID/email/something` given that we're validating an email 3/5

Reply:
Not sure for this specific endpoint but the one for verification with token, it would be breaking change. The CLI command was `mattermost user verify UUID` therefore I picked the `/user/UUID/verify` path. Maybe, we can use something like `user/UUID/email/verify?token=false`. But I'm not sure if it will be the best approach.

Reply:
Yeah. I'm not sure either. Basically we should avoid introducing breaking changes and the ideal scenario would be to have `/email/verify` for this endpoint and `/email/verifytoken` for the token one. Any idea @mgdelacroix ?

Reply:
Resolved offline, will go with `/email/verify/member` path.